src/h5web/explorer/Explorer.tsx

```diff
@@ -6,104 +6,78 @@ import React, {
 
-  function handleNodeSelect(node: TreeNode<HDF5Link>): void {
-    const isBranch = !!node.children;
-    const isExpanded = !!expandedNodes[node.uid];
-    const isSelected = node === selectedNode;
+  function handleSelect(entity: MyHDF5Entity): void {
+    const isExpanded = !!expandedGroups[entity.uid];
+    const isSelected = entity === selectedEntity;
+    onSelect(entity);
 
-    onSelect(node);
-
-    if (isBranch && (!isExpanded || isSelected)) {
-      setExpandedNodes({
-        ...expandedNodes,
-        [node.uid]: !isExpanded,
+    if (isMyGroup(entity) && (!isExpanded || isSelected)) {
+      setExpandedGroups({
+        ...expandedGroups,
+        [entity.uid]: !isExpanded,
       });
     }
   }
 
   useEffect(() => {
-    if (!tree) {
-      return;
-    }
-
-    // Get nodes on default path
-    const nodes = getNodesOnPath(tree, DEFAULT_PATH);
-
-    // Select last node in path, or root node of tree if default path was empty
-    const nodeToSelect = nodes[nodes.length - 1] || tree;
-    onSelect(nodeToSelect);
-
-    // Expand nodes on default path, excluding last node if not a branch
-    const nodesToExpand = nodeToSelect.children ? nodes : nodes.slice(0, -1);
-    setExpandedNodes(
-      nodesToExpand.reduce((acc, node) => ({ ...acc, [node.uid]: true }), {})
-    );
-  }, [onSelect, tree]);
-
-  if (!tree) {
-    return (
-      <div className={styles.explorer}>
-        <p className={styles.domain}>
-          <FiFileText className={styles.domainIcon} />
-          {domain}
-        </p>
-        <p className={styles.loading}>Loading...</p>
-      </div>
+    const entityToSelect = getEntityAtPath(root, DEFAULT_PATH);
+    onSelect(entityToSelect);
+
+    // Expand groups on default path and selected entity if group
+    setExpandedGroups(
+      entityToSelect.parents.reduce(
+        (acc, group) => ({ ...acc, [group.uid]: true }),
+        isMyGroup(entityToSelect) ? { [entityToSelect.uid]: true } : {}
+      )
     );
-  }
+  }, [onSelect, root]);
 
   return (
     <div className={styles.explorer} role="tree">
       <button
         className={styles.domainBtn}
         type="button"
         role="treeitem"
-        aria-selected={selectedNode === tree}
+        aria-selected={selectedEntity === root}
         onClick={() => {
-          onSelect(tree);
+          onSelect(root);
         }}
       >
         <FiFileText className={styles.domainIcon} />
         {domain}
       </button>
 
-      {tree.children && (
-        <TreeView
-          level={0}
-          nodes={tree.children}
-          selectedNode={selectedNode}
-          expandedNodes={expandedNodes}
-          onSelect={handleNodeSelect}
-          renderIcon={(data, isBranch, isExpanded) => (
-            <Icon data={data} isBranch={isBranch} isExpanded={isExpanded} />
-          )}
-        />
-      )}
+      <TreeView
+        level={0}
+        entities={root.children}
```

Comment:
I don't have any replacement but I find the name `entities` not extremely clear...

Reply:
Maybe it would make more sense if the component was called `EntityList`?

Reply:
Probably yes :slightly_smiling_face: